### PR TITLE
MES-4462 ADI part 2 create pages

### PR DESCRIPTION
**Pull request by edwardwilson**
## Description
Place holder pages for the ADI part 2 category.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by szabikr — APPROVED** [review]
All seems good

File: src/pages/debrief/cat-adi-part2/__tests__/debrief.cat-adi-part2.page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+import { configureI18N } from '../../../../shared/helpers/translation.helpers';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+// TODO - ADI Part 2: use correct category
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+import { FaultSummaryProvider } from '../../../../providers/fault-summary/fault-summary';
+import { of } from 'rxjs';
+import { TestOutcome } from '../../../../shared/models/test-outcome';
+import { configureTestSuite } from 'ng-bullet';
+
+describe('DebriefCatADIPart1Page', () => {
```
Comment (TomBillingtonUK) [inline src/pages/debrief/cat-adi-part2/__tests__/debrief.cat-adi-part2.page.spec.ts]: should be part 2

File: src/pages/debrief/cat-adi-part2/debrief.cat-adi-part2.page.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      select(getCurrentTest),
+    );
+    const category$ = currentTest$.pipe(
+      select(getTestCategory),
+    );
+    this.pageState = {
+      seriousFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data =>
+          this.faultSummaryProvider.getSeriousFaultsList(data, TestCategory.BE)
```
Comment (TomBillingtonUK) [inline src/pages/debrief/cat-adi-part2/debrief.cat-adi-part2.page.ts:102]: Missing TODO

File: src/pages/debrief/cat-adi-part2/debrief.cat-adi-part2.page.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+          .map(fault => fault.competencyIdentifier)),
+      ),
+      dangerousFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data =>
+          this.faultSummaryProvider.getDangerousFaultsList(data, TestCategory.BE)
```
Comment (TomBillingtonUK) [inline src/pages/debrief/cat-adi-part2/debrief.cat-adi-part2.page.ts:109]: Missing TODO

File: src/pages/debrief/cat-adi-part2/debrief.cat-adi-part2.page.ts [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+          .map(fault => fault.competencyIdentifier)),
+      ),
+      drivingFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data => this.faultSummaryProvider.getDrivingFaultsList(data, TestCategory.BE)),
```
Comment (TomBillingtonUK) [inline src/pages/debrief/cat-adi-part2/debrief.cat-adi-part2.page.ts:115]: Missing TODO

File: src/pages/health-declaration/cat-adi-part2/__tests__/health-declaration.cat-adi-part2.page.spec.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+import { MockComponent } from 'ng-mocks';
+import { Language } from '../../../../modules/tests/communication-preferences/communication-preferences.model';
+import { configureI18N } from '../../../../shared/helpers/translation.helpers';
+import { SignatureComponent } from '../../components/signature/signature';
+import { HealthDeclarationComponent } from '../../components/health-declaration/health-declaration';
+import { ReceiptDeclarationComponent } from '../../components/receipt-declaration/receipt-declaration';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { configureTestSuite } from 'ng-bullet';
+
+describe('HealthDeclarationCatBEPage', () => {
```
Comment (TomBillingtonUK) [inline src/pages/health-declaration/cat-adi-part2/__tests__/health-declaration.cat-adi-part2.page.spec.ts]: Needs updating

File: src/pages/health-declaration/cat-adi-part2/health-declaration.cat-adi-part2.page.module.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      HealthDeclarationAnalyticsEffects,
+      HealthDeclarationEffects,
+    ]),
+    HealthDeclarationComponentsModule,
+    ComponentsModule,
+    ReactiveFormsModule,
+    TranslateModule,
+  ],
+})
+export class HealthDeclarationCatBEPageModule { }
```
Comment (TomBillingtonUK) [inline src/pages/health-declaration/cat-adi-part2/health-declaration.cat-adi-part2.page.module.ts]: Incorrect class name

File: src/pages/page-names.constants.ts [hunk 7]
```diff
@@ -11,8 +11,10 @@ export const REKEY_SEARCH_PAGE = 'RekeySearchPage';
 export const TEST_RESULTS_SEARCH_PAGE = 'TestResultsSearchPage';
 export const LEGAL_REQUIREMENTS_MODAL = 'LegalRequirementsModal';
 export const REVERSE_DIAGRAM_PAGE = 'ReverseDiagramPage';
+export const REKEY_UPLOAD_OUTCOME_PAGE = 'RekeyUploadOutcomePage';
 
 export const CAT_B = {
+  REKEY_UPLOAD_OUTCOME_PAGE,
```
Comment (TomBillingtonUK) [inline src/pages/page-names.constants.ts:17]: Is there a task to TODO to go back and make sure all previous categories use the Category versions not the direct constant

File: src/pages/test-report/cat-adi-part2/components/controlled-stop/__tests__/controlled-stop.spec.ts [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+} from '../../../../../../components/common/serious-fault-badge/serious-fault-badge';
+import {
+  DangerousFaultBadgeComponent,
+} from '../../../../../../components/common/dangerous-fault-badge/dangerous-fault-badge';
+import { StartTest } from '../../../../../../modules/tests/tests.actions';
+// TO-DO ADI Part2: implement correct actions
+import {
+  ControlledStopAddDrivingFault,
+  ControlledStopRemoveFault,
+} from '../../../../../../modules/tests/test-data/cat-b/controlled-stop/controlled-stop.actions';
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-adi-part2/components/controlled-stop/__tests__/controlled-stop.spec.ts]: Controlled Stop has been made common as part of Home tests , this file no longer exists

File: src/pages/test-report/cat-adi-part2/components/controlled-stop/controlled-stop.ts [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+  hasControlledStopBeenCompleted,
+} from '../../../../../modules/tests/test-data/cat-b/test-data.cat-b.selector';
+// TO-DO ADI Part2: implement correct actions
+import {
+  ToggleControlledStop,
+  ControlledStopAddDangerousFault,
+  ControlledStopAddSeriousFault,
+  ControlledStopAddDrivingFault,
+  ControlledStopRemoveFault,
+} from '../../../../../modules/tests/test-data/cat-b/controlled-stop/controlled-stop.actions';
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-adi-part2/components/controlled-stop/controlled-stop.ts]: file no longer exists in develop

File: src/pages/test-report/cat-adi-part2/components/manoeuvres-popover/manoeuvres-popover.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+@Component({
+  selector: 'manoeuvres-popover',
+  templateUrl: 'manoeuvres-popover.html',
+})
+export class ManoeuvresPopoverComponent {
+
+  manoeuvreTypes = ManoeuvreTypes;
+  manoeuvres$: Observable<CatBUniqueTypes.Manoeuvres>;
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-adi-part2/components/manoeuvres-popover/manoeuvres-popover.ts:33]: missing TODO

File: src/pages/test-report/cat-adi-part2/components/manoeuvres/manoeuvres.ts [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+  clickCallback: OverlayCallback;
+
+  drivingFaults: number = 0;
+  hasSeriousFault: boolean = false;
+  hasDangerousFault: boolean = false;
+
+  subscription: Subscription;
+
+  displayPopover: boolean;
+  manoeuvres$: Observable<CatBUniqueTypes.Manoeuvres>;
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-adi-part2/components/manoeuvres/manoeuvres.ts:40]: missing TODO

**Review by TomBillingtonUK — COMMENTED** [review]
With the components copied across on the test report, they should really have different selectors, we have had instances where css changes on vehicle-checks component have affected others else-where in the journey

File: src/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.module.ts [hunk 12]
```diff
@@ -1,5 +0,0 @@
-import { NgModule } from '@angular/core';
```
Comment (Cavalier07) [inline src/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.module.ts]: Any reason this was deleted?

File: src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+import { LockScreenIndicator } from '../../../../components/common/screen-lock-indicator/lock-screen-indicator';
+import { CandidateSectionComponent } from '../../../../components/common/candidate-section/candidate-section';
+import { FormControl, Validators } from '@angular/forms';
+import { candidateMock } from '../../../../modules/tests/__mocks__/tests.mock';
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+import { App } from '../../../../app/app.component';
+import { MockAppComponent } from '../../../../app/__mocks__/app.component.mock';
+import { configureTestSuite } from 'ng-bullet';
+
+describe('WaitingRoomCatBEPage', () => {
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts]: needs updating

File: src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+import { InsuranceDeclarationComponent } from '../../components/insurance-declaration/insurance-declaration';
+import { ResidencyDeclarationComponent } from '../../components/residency-declaration/residency-declaration';
+import { SignatureComponent } from '../../components/signature/signature';
+import { PracticeModeBanner } from '../../../../components/common/practice-mode-banner/practice-mode-banner';
+import { EndTestLinkComponent } from '../../../../components/common/end-test-link/end-test-link';
+import { LockScreenIndicator } from '../../../../components/common/screen-lock-indicator/lock-screen-indicator';
+import { CandidateSectionComponent } from '../../../../components/common/candidate-section/candidate-section';
+import { FormControl, Validators } from '@angular/forms';
+import { candidateMock } from '../../../../modules/tests/__mocks__/tests.mock';
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts:48]: missing TODO

File: src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+  let screenOrientation: ScreenOrientation;
+  let insomnia: Insomnia;
+  let translate: TranslateService;
+  let navController: NavController;
+
+  configureTestSuite(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        WaitingRoomCatADIPart2Page,
+        MockComponent(PracticeModeBanner),
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts]: Practice Mode is not needed can be removed

File: src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+import { ScreenOrientation } from '@ionic-native/screen-orientation';
+import { ScreenOrientationMock } from '../../../../shared/mocks/screen-orientation.mock';
+import { Insomnia } from '@ionic-native/insomnia';
+import { InsomniaMock } from '../../../../shared/mocks/insomnia.mock';
+import { MockComponent } from 'ng-mocks';
+import { ConductedLanguageComponent } from '../../components/conducted-language/conducted-language';
+import { InsuranceDeclarationComponent } from '../../components/insurance-declaration/insurance-declaration';
+import { ResidencyDeclarationComponent } from '../../components/residency-declaration/residency-declaration';
+import { SignatureComponent } from '../../components/signature/signature';
+import { PracticeModeBanner } from '../../../../components/common/practice-mode-banner/practice-mode-banner';
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-adi-part2/__tests__/waiting-room.cat-adi-part2.page.spec.ts]: Can be removed

File: src/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+
+  @ViewChild(Navbar) navBar: Navbar;
+
+  pageState: WaitingRoomPageState;
+
+  formGroup: FormGroup;
+
+  subscription: Subscription;
+
+  merged$: Observable<boolean | string | CatBEUniqueTypes.JournalData>;
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts]: missing todo

File: src/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts [hunk 18]
```diff
@@ -0,0 +1,255 @@
...
+      conductedLanguage$: currentTest$.pipe(
+        select(getCommunicationPreference),
+        select(getConductedLanguage),
+      ),
+    };
+    const { welshTest$, conductedLanguage$ } = this.pageState;
+    this.merged$ = merge(
+      currentTest$.pipe(
+        select(getJournalData),
+        tap((journalData: CatBEUniqueTypes.JournalData) => {
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts]: missing TODO

File: src/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts [hunk 19]
```diff
@@ -0,0 +1,255 @@
...
+    return true;
+  }
+
+  ionViewDidLeave(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  isJournalDataInvalid = (journalData: CatBEUniqueTypes.JournalData): boolean => {
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts:202]: missiing TODO

**Review by TomBillingtonUK — COMMENTED** [review]
Another thing around components, this ticket to prep the page wasn't just meant to be a blanket copy and paste everything. It was the opportunity to make things common if they are identical between the categories. Examples like Controlled Stop and some of the stuff on the WRTC page come to mind.

File: src/pages/page-names.constants.ts (hunk 7 above)
Comment (edwardwilson) [reply src/pages/page-names.constants.ts:17]: The constant is new since all of then used the same page. This is just a tidy up to make that clear.

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**